src/GUI/UI/Houses/Shops.cpp

```diff
@@ -743,6 +664,177 @@ void GUIWindow_MagicAlchemyShop::shopWaresDialogue(bool isSpecial) {
+        if (std::find(itemClasses.begin(), itemClasses.end(), itemClass) == itemClasses.end()) {
+            itemClasses.push_back(itemClass);
+        }
+    }
+    for (int itemClass : weaponShopVariationSpecial[houseId()].item_class) {
+        if (std::find(itemClasses.begin(), itemClasses.end(), itemClass) == itemClasses.end()) {
+            itemClasses.push_back(itemClass);
+        }
+    }
+
+    for (int itemClass : itemClasses) {
+        switch (itemClass) {
+          case 23:
+            skillsOptions.push_back(DIALOGUE_LEARN_SWORD);
+            break;
+          case 24:
+            skillsOptions.push_back(DIALOGUE_LEARN_DAGGER);
+            break;
+          case 25:
+            skillsOptions.push_back(DIALOGUE_LEARN_AXE);
+            break;
+          case 26:
+            skillsOptions.push_back(DIALOGUE_LEARN_SPEAR);
+            break;
+          case 27:
+            skillsOptions.push_back(DIALOGUE_LEARN_BOW);
+            break;
+          case 28:
+            skillsOptions.push_back(DIALOGUE_LEARN_MACE);
+            break;
+          case 30:
+            skillsOptions.push_back(DIALOGUE_LEARN_STAFF);
+            break;
+          default:
+            continue;
+        }
+    }
+
+    return skillsOptions;
+}
+
+std::vector<DIALOGUE_TYPE> GUIWindow_ArmorShop::listShopLearnableSkills() {
+    std::vector<int> itemClasses;
+    std::vector<DIALOGUE_TYPE> skillsOptions;
+
+    for (int itemClass : armorShopTopRowVariationStandart[houseId()].item_class) {
+        if (std::find(itemClasses.begin(), itemClasses.end(), itemClass) == itemClasses.end()) {
+            itemClasses.push_back(itemClass);
+        }
+    }
+    for (int itemClass : armorShopBottomRowVariationStandart[houseId()].item_class) {
+        if (std::find(itemClasses.begin(), itemClasses.end(), itemClass) == itemClasses.end()) {
+            itemClasses.push_back(itemClass);
+        }
+    }
+    for (int itemClass : armorShopTopRowVariationSpecial[houseId()].item_class) {
+        if (std::find(itemClasses.begin(), itemClasses.end(), itemClass) == itemClasses.end()) {
+            itemClasses.push_back(itemClass);
+        }
+    }
+    for (int itemClass : armorShopBottomRowVariationSpecial[houseId()].item_class) {
+        if (std::find(itemClasses.begin(), itemClasses.end(), itemClass) == itemClasses.end()) {
+            itemClasses.push_back(itemClass);
+        }
+    }
+
+    for (int itemClass : itemClasses) {
+        switch (itemClass) {
+          case 31:
+            skillsOptions.push_back(DIALOGUE_LEARN_LEATHER);
+            break;
+          case 32:
+            skillsOptions.push_back(DIALOGUE_LEARN_CHAIN);
+            break;
+          case 33:
+            skillsOptions.push_back(DIALOGUE_LEARN_PLATE);
+            break;
+          case 34:
+            skillsOptions.push_back(DIALOGUE_LEARN_SHIELD);
+            break;
+          default:
+            continue;
+        }
+    }
```

Comment:
Regarding this switch --- I think it would make sense to create a single function, `itemClassToSkill`, and use it in all implementations of `listShopLearnableSkills`.